src/workerd/api/tests/compression-streams-test.js

```diff
@@ -0,0 +1,412 @@
+
+      output = await reader.read();
+    }
+
+    {
+      const { writable, readable } = new DecompressionStream('deflate-raw');
+
+      const writer = writable.getWriter();
+      const reader = readable.getReader();
+
+      await writer.write(output.value);
+      await writer.close();
+
+      const result = await reader.read();
+
+      strictEqual(dec.decode(result.value), testData);
+    }
+  },
+};
+
+// Test compression/decompression with pending read
+export const compressionPendingRead = {
+  async test() {
+    const testData = 'hello';
+    const check = new Uint8Array([
+      0x78, 0x9c, 0xcb, 0x48, 0xcd, 0xc9, 0xc9, 0x07, 0x00, 0x06, 0x2c, 0x02,
+      0x15,
+    ]);
+    const enc = new TextEncoder();
+    const dec = new TextDecoder();
+
+    {
+      const { writable, readable } = new CompressionStream('deflate');
+
+      const writer = writable.getWriter();
+      const reader = readable.getReader();
+
+      const read = reader.read();
+
+      await writer.write(enc.encode(testData));
+      await writer.close();
+
+      await read;
+    }
+
+    {
+      const { writable, readable } = new DecompressionStream('deflate');
+
+      const writer = writable.getWriter();
+      const reader = readable.getReader();
+
+      const read = reader.read();
+
+      await writer.write(check);
+      await writer.close();
+
+      const result = await read;
+
+      strictEqual(dec.decode(result.value), testData);
+    }
+  },
+};
+
+// Test cancel/abort behavior
+export const compressionCancelAbort = {
+  async test() {
+    {
+      const { writable, readable } = new CompressionStream('deflate');
+      const reader = readable.getReader();
+      const writer = writable.getWriter();
+      const promise = reader.read();
+      writer.abort(new Error('boom'));
+      await rejects(promise, { message: 'boom' });
+    }
+
+    {
+      const { readable } = new CompressionStream('deflate');
+      const reader = readable.getReader();
+      const promise = reader.read();
+      reader.cancel(new Error('boom'));
+      await rejects(promise, { message: 'boom' });
+    }
+  },
+};
+
+// Test decompression error handling
+export const decompressionError = {
+  async test() {
+    // The data is not compressed so DecompressionStream will fail.
+    const { writable, readable } = new DecompressionStream('deflate');
+
+    const writer = writable.getWriter();
+    const reader = readable.getReader();
+
+    // Write uncompressed data which should cause a decompression error
+    // The write itself may also reject, so we need to handle that too
+    const writePromise = writer
+      .write(new TextEncoder().encode('not compressed data'))
+      .catch(() => {});
+
+    // The read should fail with a TypeError
+    await rejects(reader.read(), TypeError);
+
+    // A second attempt to read also fails
+    await rejects(reader.read(), TypeError);
+
+    // Also await the write promise to prevent unhandled rejection
```

Comment:
Non-blocking nit: Comment doesn't make sense to me, but I'm not a JS expert. I'd expect `writePromise`'s `.catch()` callback to handle any rejection.